### PR TITLE
Refactor bank list item

### PR DESCRIPTION
**Pull request by nick-benoit14**
* Refactors `BankListItem` into several smaller components

### CODE REVIEW

File: client/js/_author/selectors/bank_navigator.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  common.assessments,
+  currentBankId,
+  (_rawAssessments, _currentBankId) => _.get(_rawAssessments, _currentBankId, {})
+);
+
+export const bankAssessments = createSelector(
+rawAssessments,
+common.settings,
+(_rawAssessments, _settings) => {
+  const transformedAssessments =
```
Comment (BenHSmith13) [inline client/js/_author/selectors/bank_navigator.js]: This is basically deserializing right? One day (not this day) we should do this in middleware.

Comment (nick-benoit14) [inline client/js/_author/selectors/bank_navigator.js]: Agreed, that would be a good next step. Your wisdom is much, master Ben.

File: client/js/_author/selectors/bank_navigator.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+      return assessments;
+    }, {});
+  return assessmentHash;
+});
+
+
+export const banks = createSelector(
+  path,
+  state => state.banks,
+  (_path, _banks) => {
```
Comment (BenHSmith13) [inline client/js/_author/selectors/bank_navigator.js]: So ti think I originally wrote this, which means I have no idea what it is doing. We should add a comment that says at least 'this is what makes bank navigation work', or something

Comment (nick-benoit14) [inline client/js/_author/selectors/bank_navigator.js]: Yeah I agree. I took a long time yesterday trying to figure out what it was doing ha.

**Pull request merged**